topology/mask_impl.h

```diff
@@ -87,40 +87,316 @@ Mask< D >::outside( const Box< D >& b ) const
+    Position< 2 > upper_right_sin = ( upper_right_ - cntr ) * azimuth_sin_;
+
+    double rotatedLLx = lower_left_cos[ 0 ] - lower_left_sin[ 1 ] + cntr_x;
+    double rotatedLLy = lower_left_sin[ 0 ] + lower_left_cos[ 1 ] + cntr_y;
+
+    double rotatedLRx = upper_right_cos[ 0 ] - lower_left_sin[ 1 ] + cntr_x;
+    double rotatedLRy = upper_right_sin[ 0 ] + lower_left_cos[ 1 ] + cntr_y;
+
+    double rotatedURx = upper_right_cos[ 0 ] - upper_right_sin[ 1 ] + cntr_x;
+    double rotatedURy = upper_right_sin[ 0 ] + upper_right_cos[ 1 ] + cntr_y;
+
+    double rotatedULx = lower_left_cos[ 0 ] - upper_right_sin[ 1 ] + cntr_x;
+    double rotatedULy = lower_left_sin[ 0 ] + upper_right_cos[ 1 ] + cntr_y;
+
+    double rotated_x[] = { rotatedLLx, rotatedLRx, rotatedURx, rotatedULx };
+    double rotated_y[] = { rotatedLLy, rotatedLRy, rotatedURy, rotatedULy };
+
+    min_values_[ 0 ] = *std::min_element( rotated_x, rotated_x + 4 );
+    min_values_[ 1 ] = *std::min_element( rotated_y, rotated_y + 4 );
+    max_values_[ 0 ] = *std::max_element( rotated_x, rotated_x + 4 );
+    max_values_[ 1 ] = *std::max_element( rotated_y, rotated_y + 4 );
+  }
+}
+
+template <>
+void
+BoxMask< 3 >::create_min_max_values_()
+{
+  /*
+   *        LLH      LHH
+   *       *--------*
+   *      /|       /|
+   *     / |LLL   / |LHL
+   *    *--*-----*--*
+   * HLH| /   HHH| /
+   *    |/       |/
+   *    *--------*
+   * HLL      HHL
+   *
+   */
+
+  // Rotate the corners of the box to find the new minimum and maximum x, y and
+  // z values to define the bounding box of the rotated box. We need to rotate
+  // all eight corners. If the box is not rotated, the min values corresponds
+  // to the lower_left values and the max values corresponds to the upper_right
+  // values.
+
+  if ( not is_rotated_ )
+  {
+    min_values_ = lower_left_;
+    max_values_ = upper_right_;
+  }
+  else
+  {
+    const double cntr_x = ( upper_right_[ 0 ] + lower_left_[ 0 ] ) * 0.5;
+    const double cntr_y = ( upper_right_[ 1 ] + lower_left_[ 1 ] ) * 0.5;
+    const double cntr_z = ( upper_right_[ 2 ] + lower_left_[ 2 ] ) * 0.5;
+
+    Position< 3 > cntr( cntr_x, cntr_y, cntr_z );
+
+    Position< 3 > lower_left_cos = ( lower_left_ - cntr ) * azimuth_cos_;
+    Position< 3 > lower_left_sin = ( lower_left_ - cntr ) * azimuth_sin_;
+    Position< 3 > upper_right_cos = ( upper_right_ - cntr ) * azimuth_cos_;
+    Position< 3 > upper_right_sin = ( upper_right_ - cntr ) * azimuth_sin_;
+
+    const double lower_left_polar_cos =
+      ( lower_left_[ 2 ] - cntr_z ) * polar_cos_;
+    const double lower_left_polar_sin =
+      ( lower_left_[ 2 ] - cntr_z ) * polar_sin_;
+    const double upper_right_polar_cos =
+      ( upper_right_[ 2 ] - cntr_z ) * polar_cos_;
+    const double upper_right_polar_sin =
+      ( upper_right_[ 2 ] - cntr_z ) * polar_sin_;
+
+    double rotatedLLLx =
+      ( lower_left_cos[ 0 ] - lower_left_sin[ 1 ] ) * polar_cos_
```

Comment:
@hakonsbm I decided not to do this because in addition to these two, I would also have to precalculate 6 other values ( `( upper_right_cos[ 0 ] - lower_left_sin[ 1 ] ) * polar_cos_`, ` upper_right_cos[ 0 ] - lower_left_sin[ 1 ] ) * polar_sin_`,  `( upper_right_cos[ 0 ] - upper_right_sin[ 1 ] ) * polar_cos_` etc..), and this will lead to a lot of precalculated values in addition to the several that are already in the function. As these particular values are only used twice, I think it is cleaner to leave it as is. If you still think it necessary, I can change it.